pkg/controller/servicebinding/secret_test.go

```diff
@@ -37,18 +35,96 @@ func TestSecretNew(t *testing.T) {
 	s := newSecret(
 		f.FakeDynClient(),
 		ns,
-		name,
+		secretName,
+		data,
+		secretOwnerReference,
 	)
 
-	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data, secretOwnerReference)
+	t.Run("unique_secret_name", func(t *testing.T) {
+		hashedName1, err := name(secretName, data)
+		assert.NoError(t, err)
+		assert.Contains(t, hashedName1, secretName)
+
+		hashedName2, err := name(hashedName1, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName2, hashedName1)
+	})
+
+	t.Run("check secret creation", func(t *testing.T) {
+		u, err := s.overwrite()
+		assert.NoError(t, err)
+		assertSecretNamespacedName(t, u, ns, secretName)
+		existingSecret, err := s.get()
 		assert.NoError(t, err)
-		assertSecretNamespacedName(t, u, ns, name)
+		hashedName, err := name(secretName, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName, existingSecret.GetName())
```

Comment:
what do we assert with this? Is `overwrite()` function under test? If yes, then we should have two scenarios:

* creating secret in the cluster if it does not exist
* do not create secret if it exists in the cluster with the exact content
* overwrite the secret if the version in cluster got updated externally